lib/assets/javascripts/utils/ariatiseForm.js

```diff
@@ -44,48 +44,52 @@
     3. Prevent form to be submitted
 */
 (function(ctx){
-    var requiredFields=(function(selector){
-        return $(selector).find('.form-control').filter('[aria-required="true"]');
+    // Collect all elements that are either required or have data-validation defined
+    var validatableFields=(function(selector){
+        return $(selector).find('.form-control').filter('[data-validation],[aria-required="true"]');
     });
     var blockHelp=(function (id,type){
```

Comment:
this function has turned into impure since now depends on defaultValidationError to return a value. The logic to determine what message to display should be defined outside... And returning empty string for defaultValidationError is not representative for the user. She/he could thing there is a bug.